---
layout: page
title: EEOB 563 - Links
sidebar: Links
navigation: 5
icon: <i class='fa fa-link'></i>
---

# Useful links
## General  
* [GenBank](http://www.ncbi.nih.gov/)   
* [HPC-Class](https://www.hpc.iastate.edu/guides/nova/hpc-class)

## Phylogenetic Programs
This is a small subset of available phylogenetic programs. If you are looking for more programs/options try [this list](https://phylipweb.github.io/phylip/software.html) compiled by [Joseph Felsenstein](https://felsenst.github.io/).

### Main phylogenetic programs we will use in class

* [MAFFT](https://mafft.cbrc.jp/alignment/software/) sequence alignment
* [FigTree](https://github.com/rambaut/figtree/releases) drawing phylogenetic trees
* [PAUP*](http://paup.phylosolutions.com/) Phylogenetic Analysis Using Parsimony PAUP
* [TNT](http://www.lillo.org.ar/phylogeny/tnt/) for a parsimony analysis
* [PHYLIP](https://phylipweb.github.io/phylip/), for various analysis and some obscure methods
* [FastME](http://www.atgc-montpellier.fr/fastme/) distance analysis
* [RAxML-NG](https://github.com/amkozlov/raxml-ng) Maximum Likelihood analysis
* [IQ-TREE](http://www.iqtree.org/)
* [MrBayes](https://nbisweden.github.io/MrBayes/index.html)  Bayesian analysis
* [PAML](http://abacus.gene.ucl.ac.uk/software/paml.html) hypothesis testing
* [Beast2](https://www.beast2.org/) molecular clock analysis
* [BayesTraits](https://www.evolution.reading.ac.uk/SoftwareMain.html) comparative studies

### Other programs that can be useful

* [Clustal (Omega, W and X)](http://www.clustal.org/) alternative MSA programs
* [readAI](https://vicfero.github.io/trimal/read_al.html) conversion among various MSA formats
* [TrimAI](https://vicfero.github.io/trimal/) selecting most reliable positions in an MSA
* [Gblocks](https://www.biologiaevolutiva.org/jcastresana/Gblocks.html) an older program for removing poorly aligned regions of MSA
* [PHYML](http://www.atgc-montpellier.fr/phyml/)  for a simple ML analysis
* [PhyloBayes & Coevol](http://megasun.bch.umontreal.ca/People/lartillot/www/) for a fancy Bayesian analysis
* [RevBayes](https://revbayes.github.io/) Bayesian phylogenetic inference using probabilistic graphical models and an interpreted language
* [Dendroscope](http://ab.inf.uni-tuebingen.de/software/dendroscope/) an alternative program for drawing phylogenetic trees
* [iTOL](https://itol.embl.de/) an online tool for the display, annotation and management of phylogenetic trees
* [TreeViewer](https://treeviewer.org) yet another tree-drawing program
* [Mesquite](http://mesquiteproject.wikispaces.com/)
* [TRACER](http://tree.bio.ed.ac.uk/software/tracer/) analyzing Bayesian output
* [Seq-Gen](http://tree.bio.ed.ac.uk/software/seqgen/) simulate sequence evolution on a phylogeny

Systematics/Phylogenetics Sites
---
* [Open Tree of Life](https://tree.opentreeoflife.org/)
* [Interactive Tree of Life](http://itol.embl.de/)
* [Timetree of Life](http://www.timetree.org/)


Blogs
---
* [Doing Bayesian Data Analysis](http://doingbayesiandataanalysis.blogspot.com/)
* [The Bayesian kitchen](http://bayesiancook.blogspot.com/)
