---
layout: page
title: EEOB 563 - Schedule
sidebar: Schedule
navigation: 6
icon: <i class='fa fa-calendar'></i> 
---

# Schedule

I will update this page before every class with the links to in-class materials and assignments. 
To view the planned schedule of course activities, view the class calendar using the button below.

<a href="https://docs.google.com/spreadsheets/d/1okuP20eZHE3TKdtxlqVbEp3xEcYB6wGZh_O1muiiXb0/edit?usp=sharing"><button type="button" class="btn btn-primary">Class Calendar</button></a>

### Week 1
* Lecture 1: Introduction to Molecular Phylogenetics [[slides](https://isu-molphyl.github.io/EEOB563-Spring2020/lecture_notes/00_Introduction.pdf)]
	* [Assignment 1](./assignments.md) (**due 1/16**)

* Lecture 2: Trees and characters [[notes](https://isu-molphyl.github.io/EEOB563-Spring2020/lecture_notes/01_Phylogenetic_trees.pdf)]
    * [Assignment 1](./assignments.md) is due.
    * Reading: Allman and Rhodes (2016).  Chapter 2: Combinatorics of Trees I.  
    * Discussion: Baum et al. (2005). The Tree-Thinking Challenge.  [Science 310: 979-980](http://science.sciencemag.org/content/310/5750/979.full.pdf).  

### Week 2
* Lecture 3: Homology and Sequence Alignment [[notes](https://isu-molphyl.github.io/EEOB563-Spring2020/lecture_notes/02_Homology_and_alignments.pdf)]
    * Reading: Lemey, Salemi and Vandamme, [Chapter 3: 3.1-3.10](./chapter3.pdf).  
    * Discussion: Fitch (2000). Homology a personal view on some of the problems. [TIG 16: 227](https://doi.org/10.1016/S0168-9525(00)02005-9).
* [Computer Lab 1](https://isu-molphyl.github.io/EEOB563/computer_labs/lab1): Introduction to Unix/GitHub/SSH
    * Complete all [prerequisites](https://isu-molphyl.github.io/EEOB563/computer_labs/lab1) before coming to the class! 
    * Post any question you may have on Slack.
    * [Assignment 2](https://isu-molphyl.github.io/EEOB563-Spring2020/assignments/assignment2.pdf) (**due 01/30**)

### Week 3  
* Lecture 4: Parsimony and cladistics. Optimality criteria and character optimization.  
    * Reading:  Allman and Rhodes (2016).  Chapter 3: Parsimony. [[notes](https://isu-molphyl.github.io/EEOB563-Spring2020/lecture_notes/03_Maximum_parsimony.pdf)]
* Lecture 5: Searching tree space. Measures of character fit. Assessing clade support.  
    * Reading: Allman and Rhodes (2016).  Chapter 9: Tree Space. [[notes](https://isu-molphyl.github.io/EEOB563-Spring2020/lecture_notes/03_Maximum_parsimony.pdf)]
    * Discussion (leader _Phillip_): Baron et al. (2017). A new hypothesis of dinosaur relationships and early dinosaur evolution. [Nature 543: 501-506](https://www.nature.com/articles/nature21700). 
    See also [News & Views](https://www.nature.com/articles/543494a), Langer et al. [communication](https://www.nature.com/articles/nature24011), and Baron et al. [reply](https://www.nature.com/articles/nature24012).

### Week 4  
* [Computer Lab 2](https://isu-molphyl.github.io/EEOB563/computer_labs/lab2): Multiple Sequence Alignment and Parsimony analysis.
    * All users, make sure you have a GitHub account!
    * You may want to install ClustalW, Mafft, PAUP\*, and TNT on your computer [[links](https://isu-molphyl.github.io/EEOB563/links)]. However, we will use HPC-class cluster for the excercises.
    * [Assignment 3](https://isu-molphyl.github.io/EEOB563-Spring2020/assignments/assignment3.pdf) (**due 02/11**)
* Lecture 6: Distance matrix methods. Clustering algorithms. [[notes](https://isu-molphyl.github.io/EEOB563-Spring2019/lecture_notes/04_Distance_methods.pdf)]
    * Reading: Allman and Rhodes (2016).  Chapter 5: Distance Methods.
    * Discussion (leader _Jennifer_): Naxerova et al. (2017). Origins of lymphatic and distant metastases in human colorectal cancer. [Science 357, 55-60](http://science.sciencemag.org/content/357/6346/55).

### Week 5 (Feb 11, 13)  
* Lecture 7: NJ algorithm; Model-based distances. [[notes](https://isu-molphyl.github.io/EEOB563-Spring2020/lecture_notes/04_Distance_methods.pdf)]
    * Reading: Reading: Allman and Rhodes (2016).  Chapter 6: Probabilistic Models of DNA Mutations.
* [Computer Lab 3](https://isu-molphyl.github.io/EEOB563/computer_labs/lab3): Distance analysis with PAUP, PHYLIP, and FastME.  
    * [Assignment 4](https://isu-molphyl.github.io/EEOB563-Spring2020/assignments/assignment4.pdf) (**due 02/20**)

<!--
* Reading: Reading: Allman and Rhodes (2016).  Chapter 7: Model-based Distances.
-->

### Week 6 (Feb 18, 20) 
* Lecture 8: Introduction to Maximum Likelihood. [[notes](https://isu-molphyl.github.io/EEOB563-Spring2020/lecture_notes/05_Maximum_likelihood.pdf)]
    * Reading: Allman and Rhodes (2016).  Chapter 8: Maximum Likelihood.
* Lecture 9: Constructing Phylogenetic Trees Using Maximum Likelihood. [[notes](https://isu-molphyl.github.io/EEOB563-Spring2020/lecture_notes/05_Maximum_likelihood.pdf)]
    * Discussion (leader _Alexandra_): Zhang, L., Chen, F., Zhang, X. et al. (2020). The water lily genome and the early evolution of flowering plants. [Nature 577, 79–84](https://doi.org/10.1038/s41586-019-1852-5).
    See also [Research Highlight](https://www.nature.com/articles/s41477-019-0587-5)

### Week 7 (Feb 25, 27)
* [Computer Lab 4](https://isu-molphyl.github.io/EEOB563/computer_labs/lab4): Likelihood analysis in FastML and RAxML.
    * [Assignment 5](https://isu-molphyl.github.io/EEOB563-Spring2020/assignments/assignment5.pdf) (**due 03/05**)
* Lecture 10: Bayes’ theorem and Bayesian methods in phylogenetics. [[notes](https://isu-molphyl.github.io/EEOB563-Spring2020/lecture_notes/06_Bayesian_phylogenetics.pdf)]
    * Reading: Allman and Rhodes (2016).  Chapter 12: Bayesian Inference.

### Week 8 (Mar 3, 5)
* Lecture 11: Applications of Bayesian methods. [[notes](https://isu-molphyl.github.io/EEOB563-Spring2020/lecture_notes/06_Bayesian_phylogenetics.pdf)]
    * Discussion (leader _Leila_): Williams et al., 2020. Phylogenomics provides robust support for a two-domains tree of life. [Nature Ecology and Evoluiton, 4:138–147](https://isu-molphyl.github.io/EEOB563-Spring2020/discussion_articles/williams2020_42635.pdf)

* [Computer Lab 5](https://isu-molphyl.github.io/EEOB563-Spring2020/computer_labs/lab5): Bayesian analysis with MrBayes.
     * [Assignment 6](https://isu-molphyl.github.io/EEOB563-Spring2020/assignments/assignment6.pdf) (**due 03/26**)[[dataset]](https://isu-molphyl.github.io/EEOB563-Spring2020/assignments/hiv.nxs)

### Week 9 (Mar 10, 12)
* Midterm review
    * Reading: Yang and Rannala (2012). Molecular phylogenetics: principles and practice. [Nature Review Genetics, 13:303-314](https://www.nature.com/articles/nrg3186.pdf)

**03/12 Midterm Exam**

### Week 10: Spring Break! 
* Don’t forget about your [final project](./final_project.md) outline! (**due 3/24**)

### Week 11 (Mar 24, 26) -- COVID-19 edition
* Topic 12: Model selection and model averaging in Likelihood and Bayesian methods [[notes](https://isu-molphyl.github.io/EEOB563-Spring2020/lecture_notes/07_Hypothesis_testing.pdf)].
    * In lieu of the lecture, read: Sullivan and Joyce 2005.  Model Selection in Phylogenetics 
    [Annnu. Rev. Ecol. Evol. Syst, 36:445–466](https://www.annualreviews.org/doi/abs/10.1146/annurev.ecolsys.36.102003.152633).
    * In lieu of the discussion, read: Abadi et al. 2019. Model selection may not be a mandatory step for
    phylogeny reconstruction. [Nature Communications 10:934](https://doi.org/10.1038/s41467-019-08822-w)
    * Post at least one comment question in the reading-discussion channel on Slack and answer/comment on another post.
* Topic 13: Neutral and adaptive protein evolution [[notes](https://isu-molphyl.github.io/EEOB563-Spring2020/lecture_notes/08_Molecular_evolution.pdf)].
    * In lieu of the lecture, read: Bielawski and Yang (2000). Statistical methods for detecting molecular adaptation.  [TREE 15(12): 496-503](http://www.cell.com/trends/ecology-evolution/abstract/S0169-5347(00)01994-7).
    * For additional details and examples, read (optional): Yang 2019. Adaptive Molecular Evolution. In. Balding et al. (Eds). [Handbook of Statistical Genomics, 4ed.](https://doi.org/10.1002/9781119487845.ch13).  
    * In lieu of the discussion, read: Tenaillon et al. (2016). Tempo and mode of genome evolution in a 50,000-generation experiment. [Nature 536: 165-170](https://www.nature.com/articles/nature18959).
    * Post at least one comment question in the reading-discussion channel on Slack and answer/comment on another post.

### Week 12 (Mar 31, Apr 2) -- COVID-19 edition
* [Computer Lab 6](https://isu-molphyl.github.io/EEOB563/computer_labs/lab6): Hypotheses testing with PAML.
    * I will give a short ZOOM intro and will be around (virtually) to help.  
* Topic 14: Gene trees and species trees [[notes](https://isu-molphyl.github.io/EEOB563-Spring2020/lecture_notes/09_Gene_trees_species_trees.pdf)].
    * Reading: Maddison (1997). Gene Trees in Species Trees. [Systematic Biology 46: 523–536](https://doi.org/10.1093/sysbio/46.3.523).
    * Discussion: Gatesy and Springer (2014). Phylogenetic analysis at deep timescales: 
    Unreliable gene trees, bypassed hidden support, and the coalescence/concatalescence conundrum. 
    [Molecular Phylogenetics and Evolution 80: 231-266](https://doi.org/10.1016/j.ympev.2014.08.013)A

### Week 13 (Apr 7, 9) -- COVID-19 edition
* Topic 15: Timing the evolutionary events [[notes](https://isu-molphyl.github.io/EEOB563-Spring2018/lecture_notes/10_Molecular_clocks.pdf)].
    * In lieu of the lecture, read: Sauquet (2013). A practical guide to molecular dating. [C. R. Palevol 12. 355–367](https://www.sciencedirect.com/science/article/pii/S1631068313001097)
    * Here is a newer paper that I find very useful: Bromham et al. (2018). Bayesian molecular dating: opening up the black box. [Biological Reviews, 93: 1165-1191](https://doi.org/10.1111/brv.12390)
    * Discussion: Worobey (2008). Direct evidence of extensive diversity of HIV-1 in Kinshasa by 1960. 
    [Nature 455: 661–664](https://doi.org/10.1038/nature07390). A cool paper that involves molecular clock analyses and paleovirology.
    * Special discussion paper (optional): Tang et al. (2020). On the origin and continuing evolution of SARS-CoV-2. [National Science Review, nwaa036](https://doi.org/10.1093/nsr/nwaa036). *Note:* This paper is not directly related to the present topic, but shows application of 
    several techniques we discussed in class.
    * Remember to post at least one comment/question on either paper in the reading-discussion channel on Slack and answer/comment on another post.
* [Computer Lab 7](https://isu-molphyl.github.io/EEOB563/computer_labs/lab7): Taming the BEAST.

### Week 14 (Apr 14, 16) -- COVID-19 edition
* Topic 16: Ancestral State Reconstruction and Comparative tests [[notes](https://isu-molphyl.github.io/EEOB563-Spring2018/lecture_notes/04_10_18.pdf)].
    * In lieu of the lecture, read: Joy et al. 2016. Ancestral reconstruction. [PLoS Comput. Biol. 12(7): e1004763](https://doi.org/10.1371/journal.pcbi.1004763) 
    * Discussion: Crisp and Cook (2005). Do early branching lineages signify ancestral traits? [Trends Ecol. Evol. 20:122-128](https://doi.org/10.1016/j.tree.2004.11.010)
    
* [Computer Lab 8](https://isu-molphyl.github.io/EEOB563/computer_labs/lab8): BayesTraits.
    * In lieu of the introduction, read: Pagel and Meade (2006).  Bayesian Analysis of Correlated Evolution of Discrete Characters by  Reversible-Jump Markov Chain Monte Carlo.  [Am. Nat. 167:808-825](https://doi.org/10.1086/503444).
    * **Final project draft due:** share your GitHub/GitLab address with two assigned reviewers.  Perform the kind of positive, constructive review you would like to get on your own draft.  Prepare your reviews by 4/21.  

### Week 15 (Apr 21, 23) -- COVID-19 edition
* Topic 17: Phylogenomics and the tree of life [[notes](https://isu-molphyl.github.io/EEOB563-Spring2018/lecture_notes/04_10_18.pdf)].
    * In lieu of the lecture, read: Delsuc et al. 2005.  Phylogenomics and the reconstruction of the tree of life.  [Nature Reviews Genetics.  6: 361-375](https://www.nature.com/articles/nrg1603). (This is old now, but rather classical).
    * Discussion: Simion et al. (2017). A Large and Consistent Phylogenomic Dataset Supports Sponges as the Sister Group to All Other Animals. [Current biology 2: 958-967](https://www.sciencedirect.com/science/article/pii/S0960982217301999).

<!--


* Topic 18: Signal vs. noise in phylogenetic reconstruction [[notes](https://isu-molphyl.github.io/EEOB563-Spring2018/lecture_notes/04_10_18.pdf)].
    * Discussion: Philippe et al. 2017. Pitfalls in supermatrix phylogenomics. [European Journal of Taxonomy 283: 1-25](http://www.europeanjournaloftaxonomy.eu/index.php/ejt/article/view/407).

    
* **Revised final project due on 4/28 at 11:59pm (commit to GitHub and email me the link)**

### Final Presentations [[Scoring rubric](./scoring_rubric.pdf)]:
* 04/30: 9:00-10:50am
  * Elizabeth
  * Lauren  
  * Jing 
  * Jermaine
  * Zakayo
* 05/02: 9:00-10:50am
  * Shofi
  * Jade
  * Amruta
  * Jordan
  * Dennis
* 05/07: 8:00-9:30am
  * Katelyn
  * Devin
  * Christian
  * Zachary

### Week 11 (Mar 24, 26) -- Coronavirus edition
* Lecture 12: Model selection and model averaging in Likelihood and Bayesian methods.
    * Reading and Discussion: Posada and Buckley 2004.  Model selection and model averaging in phylogenetics:
    advantages of Akaike Information Criterion and Bayesian Approaches over Likelihood Ratio tests. [Systematic Biology, 53:793?808](https://academic.oup.com/sysbio/article/53/5/793/2842928).
    

* Midterm exam review and [final project](./final_project.md) discussion
    * Be ready to present your final project outline.  Include hypotheses, data, and proposed methods for the project. In addition, create a GitHub/GitLab repository for the final project and send me the
link. 

* [Computer Lab 5](https://isu-molphyl.github.io/EEOB563/computer_labs/lab5): Bayesian analysis with MrBayes.
  * [Assignment 6](https://isu-molphyl.github.io/EEOB563-Spring2020/assignments/assignment6.pdf) (**due 04/04**) [[dataset]](https://isu-molphyl.github.io/EEOB563-Spring2020/assignments/hiv.nxs)


Lecture 13 (ha!): Gene trees and species trees.
    * Reading: Allman and Rhodes (2016).  Chapter 13: Gene trees and species trees
    * Discussion: Copetti et al. 2017. Extensive gene tree discordance and hemiplasy shaped the genomes of North American columnar cacti. [Proc Natl Acad Sci U S A: 114: 12003-12008](http://www.pnas.org/   
content/114/45/12003.full).
-->
<!--
### Week 12 
* Lecture 14: Neutral and adaptive protein evolution [[notes](https://isu-molphyl.github.io/EEOB563-Spring2020/lecture_notes/08_Molecular_evolution.pdf)].
    * Reading: Bielawski and Yang (2000). Statistical methods for detecting molecular adaptation.  [TREE 15(12): 496-503](http://www.cell.com/trends/ecology-evolution/abstract/S0169-5347(00)01994-7).
    * Discussion (leaders _Jade and Zachary_):  Tenaillon et al. (2016). Tempo and mode of genome evolution in a 50,000-generation experiment. [Nature 536: 165-170](https://www.nature.com/articles/         
nature18959).

* [Computer Lab 6](https://isu-molphyl.github.io/EEOB563/computer_labs/lab6): Hypotheses testing with PAML.

### Week 13
* Lecture 15: Timing the evolutionary events [[notes](https://isu-molphyl.github.io/EEOB563-Spring2018/lecture_notes/04_03_18.pdf)].
    * Reading: Sauquet (2013). A practical guide to molecular dating. [C. R. Palevol 12. 355?367](https://www.sciencedirect.com/science/article/pii/S1631068313001097)
<!--    * Discussion: Worobey et al. (2016). 1970s and ?Patient 0? {HIV}-1 genomes illuminate early {HIV}/{AIDS} history in North America. [Nature 539: 98-101](https://www.nature.com/articles/nature19827).

* [Computer Lab 7](https://isu-molphyl.github.io/EEOB563/computer_labs/lab7): Taming the BEAST.

### Week 14 
* Lecture 16: Ancestral State Reconstruction and Comparative tests [[notes](https://isu-molphyl.github.io/EEOB563-Spring2018/lecture_notes/04_10_18.pdf)].
    * Reading: Pagel and Meade (2006).  Bayesian Analysis of Correlated Evolution of Discrete Characters by Reversible-Jump Markov Chain Monte Carlo.  Am. Nat. 167:808-825.
    * Discussion: (leaders _Katelyn and Jermaine_) Watts et al. 2016. Ritual human sacrifice promoted and sustained the evolution of stratified societies. [Nature 532: 228-231](https://www.nature.com/      
articles/nature17159).

* [Computer Lab 8](https://isu-molphyl.github.io/EEOB563/computer_labs/lab8): BayesTraits.
    * Reading (again): Pagel and Meade (2006).  Bayesian Analysis of Correlated Evolution of Discrete Characters by  Reversible-Jump Markov Chain Monte Carlo.  Am. Nat. 167:808-825.
    * **Final project draft due:** share your GitHub/GitLab address with two assigned reviewers.  Perform the kind of positive, constructive review you would like to get on your own draft.  Prepare your    
reviews by 4/23.  

### Week 15 (Current and Last!)
* Lecture 17: Phylogenomics and the tree of life [[notes](https://isu-molphyl.github.io/EEOB563-Spring2018/lecture_notes/04_10_18.pdf)].
    * Reading: Delsuc et al. 2005.  Phylogenomics and the reconstruction of the tree of life.  [Nature Reviews Genetics.  6: 361-375](https://www.nature.com/articles/nrg1603).
    * Discussion: Simion et al. (2017). A Large and Consistent Phylogenomic Dataset Supports Sponges as the Sister Group to All Other Animals. [Current biology 2: 958-967](https://www.sciencedirect.com/    
science/article/pii/S0960982217301999).

* Lecture 18: Signal vs. noise in phylogenetic reconstruction [[notes](https://isu-molphyl.github.io/EEOB563-Spring2018/lecture_notes/04_10_18.pdf)].
    * Discussion: Philippe et al. 2017. Pitfalls in supermatrix phylogenomics. [European Journal of Taxonomy 283: 1-25](http://www.europeanjournaloftaxonomy.eu/index.php/ejt/article/view/407).

    
* **Revised final project due on 4/28 at 11:59pm (commit to GitHub and email me the link)**

<!---  --->
