---
layout: page
title: EEOB 563 - Schedule
sidebar: Schedule
navigation: 2
icon: <i class='fa fa-calendar'></i>
---

# Schedule

I will update this page before every class with the links to in-class materials and assignments.
To view the planned schedule of course activities, view the class calendar using the button below.

<a href="https://docs.google.com/spreadsheets/d/1okuP20eZHE3TKdtxlqVbEp3xEcYB6wGZh_O1muiiXb0/edit?usp=sharing"><button type="button" class="btn btn-primary">Class Calendar</button></a>

### Week 1 (Jan 17, 19)
* Lecture 1: Introduction to Molecular Phylogenetics [[slides](https://isu-molphyl.github.io/EEOB563-Spring2023/lecture_notes/00_Introduction.pdf)]
	* [Assignment 1](./assignments.md) (**due 1/19**)
* Lecture 2: Trees and characters [[notes](https://isu-molphyl.github.io/EEOB563-Spring2021/lecture_notes/01-phylogenetic_trees.pdf)]
    * [Assignment 1](./assignments.md) is due.
    * Reading: Allman and Rhodes (2016).  Chapter 2: Combinatorics of Trees I.  
    * Discussion: Baum et al. (2005). The Tree-Thinking Challenge.  [Science 310: 979-980](http://science.sciencemag.org/content/310/5750/979.full.pdf).  

### Week 2 (Jan 24, 26)
* Lecture 3: Homology and Sequence Alignment [[notes](https://isu-molphyl.github.io/EEOB563-Spring2021/lecture_notes/02-homology_and_msa.pdf)]
    * Reading: Lemey, Salemi and Vandamme, [Chapter 3: 3.1-3.10](./chapter3.pdf).  
    * Discussion: Fitch (2000). Homology a personal view on some of the problems. [TIG 16: 227](https://doi.org/10.1016/S0168-9525(00)02005-9).
* [Computer Lab 1](https://isu-molphyl.github.io/EEOB563/computer_labs/lab1): Introduction to Unix/Multiple Sequence Alignment (MSA).
    * Complete all [prerequisites](https://isu-molphyl.github.io/EEOB563/computer_labs/lab1) before coming to the class!
    * Post any question you may have on Slack.
    * [Assignment 2](https://isu-molphyl.github.io/EEOB563-Spring2021/assignments/assignment2.pdf) (**due 02/02**)

<!--
### Week 3 (Jan 31, Feb 2)
* Lecture 4: Parsimony and cladistics. Optimality criteria and character optimization.  
    * Reading:  Allman and Rhodes (2016).  Chapter 3: Parsimony. [[notes](https://isu-molphyl.github.io/EEOB563-Spring2021/lecture_notes/03_Maximum_parsimony.pdf)]
* Lecture 5: Searching tree space. Measures of character fit. Assessing clade support.  
    * Reading: Allman and Rhodes (2016).  Chapter 9: Tree Space. [[notes](https://isu-molphyl.github.io/EEOB563-Spring2021/lecture_notes/03_Maximum_parsimony.pdf)]
    * Discussion: Baron et al. (2017). A new hypothesis of dinosaur relationships and early dinosaur evolution. [Nature 543: 501-506](https://www.nature.com/articles/nature21700).
    See also [News & Views](https://www.nature.com/articles/543494a), Langer et al. [communication](https://www.nature.com/articles/nature24011), and Baron et al. [reply](https://www.nature.com/articles/nature24012).

### Week 4 (Feb 16, 18)
* [Computer Lab 2](https://isu-molphyl.github.io/EEOB563/computer_labs/lab2): Multiple Sequence Alignment and Parsimony analysis.
    * All users, make sure you have a GitHub account!
    * You may want to install ClustalW, Mafft, PAUP\*, and TNT on your computer [[links](https://isu-molphyl.github.io/EEOB563/links)]. However, we will use HPC-class cluster for the excercises.
    * [Assignment 3](https://isu-molphyl.github.io/EEOB563-Spring2021/assignments/assignment3.pdf) (**due 02/25**)
* Lecture 6: Distance matrix methods. Clustering algorithms. [[notes](https://isu-molphyl.github.io/EEOB563-Spring2021/lecture_notes/04_Distance_methods.pdf)]
    * Reading: Allman and Rhodes (2016).  Chapter 5: Distance Methods.
    * Discussion: Naxerova et al. (2017). Origins of lymphatic and distant metastases in human colorectal cancer. [Science 357, 55-60](http://science.sciencemag.org/content/357/6346/55).

### Week 5 (Feb 23, 25)  
* Lecture 7: NJ algorithm; Model-based distances. [[notes](https://isu-molphyl.github.io/EEOB563-Spring2021/lecture_notes/04_Distance_methods.pdf)]
    * Reading: (Optional!) Allman and Rhodes (2016).  Chapter 6: Probabilistic Models of DNA Mutations.
* [Computer Lab 3](https://isu-molphyl.github.io/EEOB563/computer_labs/lab3): Distance analysis with PAUP, PHYLIP, and FastME.  
    * [Assignment 4](https://isu-molphyl.github.io/EEOB563-Spring2021/assignments/assignment4.pdf) (**due 03/04**)

* Reading: Reading: Allman and Rhodes (2016).  Chapter 7: Model-based Distances.

### Week 6 (Mar 2, 4)
* Lecture 8: Introduction to Maximum Likelihood. [[notes](https://isu-molphyl.github.io/EEOB563-Spring2021/lecture_notes/05_Maximum_likelihood.pdf)]
    * Reading: Allman and Rhodes (2016).  Chapter 8: Maximum Likelihood.
* Lecture 9: Constructing Phylogenetic Trees Using Maximum Likelihood. [[notes](https://isu-molphyl.github.io/EEOB563-Spring2021/lecture_notes/05_Maximum_likelihood.pdf)]
    * Discussion: Basem Al-Shayeb et al. (2020). Clades of huge phages from across Earth’s ecosystems. [Nature 578, 425–431](https://doi.org/10.1038/s41586-020-2007-4).
    If you want to brush up your knowledge on bacteriophages, here is a nice [review](https://doi.org/10.1038/s41586-019-1894-8) from the same issue of Nature.

    * Discussion: Hug at al. (2016). A new view of the tree of life. [Nat. Microb. 1: 16048](https://doi.org/10.1038/nmicrobiol.2016.48)

#### &#10071; It's a right time to start working on your final project! &#10071;

Here are the [instructions](./final_project.md).

### Week 7 (Mar 9, 11)
* [Computer Lab 4](https://isu-molphyl.github.io/EEOB563/computer_labs/lab4): Likelihood analysis in FastML and RAxML.
    * [Assignment 5](https://isu-molphyl.github.io/EEOB563-Spring2021/assignments/assignment5.pdf) (**due 03/16**)
* Lecture 10: Bayes’ theorem and Bayesian methods in phylogenetics. [[notes](https://isu-molphyl.github.io/EEOB563-Spring2021/lecture_notes/06_Bayesian_phylogenetics.pdf)]
    * Reading: Allman and Rhodes (2016).  Chapter 12: Bayesian Inference.

### Week 8 (Mar 16, 18)
* Lecture 11: Applications of Bayesian methods. [[notes](https://isu-molphyl.github.io/EEOB563-Spring2021/lecture_notes/06_Bayesian_phylogenetics.pdf)]
    * Discussion: Williams et al., 2020. Phylogenomics provides robust support for a two-domains tree of life. [Nature Ecology and Evoluiton, 4:138–147](https://isu-molphyl.github.io/EEOB563-Spring2021/discussion_articles/williams2020_42635.pdf)
* Midterm review.

* [Computer Lab 5](https://isu-molphyl.github.io/EEOB563-Spring2021/computer_labs/lab5): Bayesian analysis with MrBayes.
     * [Assignment 6](https://isu-molphyl.github.io/EEOB563-Spring2021/assignments/assignment6.pdf) (**due 03/25**)[[dataset]](https://isu-molphyl.github.io/EEOB563-Spring2021/assignments/hiv.nxs)

#### &#10071; Midterm exam &#10071;
- Exam will open on Mar 18, after the class. Will cover weeks 1-7
- Exam will close on Mar 23 before the class.
- You will have one 3hr window to complete the exam.


### Week 10: Spring Break!
* Don’t forget about your [final project](./final_project.md) outline! (**due 3/24**)

### Week 9 (Mar 23, 25)
* [Computer Lab 5](https://isu-molphyl.github.io/EEOB563-Spring2021/computer_labs/lab5): Bayesian analysis with MrBayes.
     * [Assignment 6](https://isu-molphyl.github.io/EEOB563-Spring2021/assignments/assignment6.pdf) (**due 04/01**)[[dataset]](https://isu-molphyl.github.io/EEOB563-Spring2021/assignments/hiv.nxs)
* Lecture 12: Model selection and model averaging in Likelihood and Bayesian methods [[notes](https://isu-molphyl.github.io/EEOB563-Spring2021/lecture_notes/07_Model_use.pdf)].
    * Reading: Posada & Buckley (2004).  Model Selection and Model Averaging in
		Phylogenetics: Advantages of Akaike Information Criterion and Bayesian
		Approaches Over Likelihood Ratio Tests.
    [Systematic Biology 53: 793–808](https://doi.org/10.1080/10635150490522304).  
			* Alternative (simpler) reading: Kelchner & Thomas (2007). Model use in phylogenetics: nine key questions. [TREE 22: 87-94](https://doi.org/10.1016/j.tree.2006.10.004)
    * Discussion: Abadi et al. 2019. Model selection may not be a mandatory step for
    phylogeny reconstruction. [Nature Communications 10:934](https://doi.org/10.1038/s41467-019-08822-w)  
			* There is a critique in [BioRxiv](https://doi.org/10.1101/849018) if you are interested.  
    	* Remember to post at least one comment question in the reading-discussion channel on Slack!  

### Week 10 (Mar 30, Apr 1)
* Lecture 13: Neutral and adaptive protein evolution [[notes](https://isu-molphyl.github.io/EEOB563-Spring2021/lecture_notes/08_Molecular_evolution.pdf)].
    * Reading: Bielawski and Yang (2000). Statistical methods for detecting molecular adaptation.  [TREE 15(12): 496-503](http://www.cell.com/trends/ecology-evolution/abstract/S0169-5347(00)01994-7).
    * For additional details and examples, read (optional): Yang 2019. Adaptive Molecular Evolution. In. Balding et al. (Eds). [Handbook of Statistical Genomics, 4ed.](https://doi.org/10.1002/9781119487845.ch13).  
    * Discussion: Tenaillon et al. (2016). Tempo and mode of genome evolution in a 50,000-generation experiment. [Nature 536: 165-170](https://www.nature.com/articles/nature18959).  
    * Remember to post a question/comment in the reading-discussion channel on Slack!  
* [Computer Lab 6](https://isu-molphyl.github.io/EEOB563/computer_labs/lab6): Hypotheses testing with PAML.

### Week 11 (Apr 6, 8)
* Lecture 14: Gene trees and species trees [[notes](https://isu-molphyl.github.io/EEOB563-Spring2021/lecture_notes/09_Gene_species_trees.pdf)].  
    * Reading: Maddison (1997). Gene Trees in Species Trees. [Systematic Biology 46: 523–536](https://doi.org/10.1093/sysbio/46.3.523).  
    * Discussion: Gatesy and Springer (2014). Phylogenetic analysis at deep timescales:
    Unreliable gene trees, bypassed hidden support, and the coalescence/concatalescence conundrum.
    [Molecular Phylogenetics and Evolution 80: 231-266](https://doi.org/10.1016/j.ympev.2014.08.013)  
* Lecture 15: Phylogenomics and the tree of life  [[notes](https://isu-molphyl.github.io/EEOB563-Spring2021/lecture_notes/10_Phylogenomics.pdf)].  
		* Reading: Young and Gillung (2020). Phylogenomics – principles, opportunities and pitfalls of big-data phylogenetics. [Systematic Entomology 45: 225–247](https://doi.org/10.1111/syen.12406)  
    * Discussion: Simion et al. (2021). To What Extent Current Limits of Phylogenomics Can Be Overcome?.
		In Scornavacca et al., editors, [Phylogenetics in the Genomic Era, chapter No. 2.1, pp.  2.1:1–2.1:34](https://hal.archives-ouvertes.fr/hal-02535366/document).    

### Week 12 (Apr 13, 15)
* [Computer Lab 7](https://isu-molphyl.github.io/EEOB563/computer_labs/lab7): Phylogenomics.
* Lecture 16: Ancestral State Reconstruction [[notes](https://isu-molphyl.github.io/EEOB563-Spring2021/lecture_notes/11_Ancestral_reconstruction.pdf)].
    * Reading: Joy et al. 2016. Ancestral reconstruction. [PLoS Comput. Biol. 12(7): e1004763](https://doi.org/10.1371/journal.pcbi.1004763)
    * Discussion: Crisp and Cook (2005). Do early branching lineages signify ancestral traits? [Trends Ecol. Evol. 20:122-128](https://doi.org/10.1016/j.tree.2004.11.010)
    *  &#10071; Final project draft due:  &#10071; share your GitHub/GitLab address with two assigned reviewers.  Perform the kind of positive, constructive review you would like to get on your own draft.  Prepare your reviews by 4/22.  

### Week 13 (Apr 20, 22)
* Lecture 17: Phylogenetic comparative methods [[notes](https://isu-molphyl.github.io/EEOB563-Spring2021/lecture_notes/12_Phylogenetic_comparative_methods)].
    * Reading: Uyeda et al. 2018. Rethinking phylogenetic comparative methods. [Systematic Biology 67: 1091–1109](https://doi.org/10.1093/sysbio/syy031)
    * Discussion: Watts et al. 2016. Ritual human sacrifice promoted and sustained the evolution of stratified societies. [Nature 532: 228-231](https://www.nature.com/articles/nature17159).
* [Computer Lab 8](https://isu-molphyl.github.io/EEOB563/computer_labs/lab8): BayesTraits.
    * Introductory reading: Pagel and Meade (2006).  Bayesian Analysis of Correlated Evolution of Discrete Characters by  Reversible-Jump Markov Chain Monte Carlo.  [Am. Nat. 167:808-825](https://doi.org/10.1086/503444).

Add: https://ecoevorxiv.org/ytm5x/ (Phylogenetic Comparative Methods: A User’s Guide for Paleontologists
)

### Week 14 (Apr 27, 29)

* Lecture 18: Timing the evolutionary events [[notes](https://isu-molphyl.github.io/EEOB563-Spring2021/lecture_notes/13_Molecular_clocks.pdf)].
    * Reading: Bromham et al. (2018). Bayesian molecular dating: opening up the black box. [Biological Reviews, 93: 1165-1191](https://doi.org/10.1111/brv.12390)
    * Discussion: Worobey (2008). Direct evidence of extensive diversity of HIV-1 in Kinshasa by 1960.
    [Nature 455: 661–664](https://doi.org/10.1038/nature07390). A cool paper that involves molecular clock analyses and paleovirology.
      * Remember to post at least one comment/question on either paper in the reading-discussion channel on Slack and answer/comment on another post.

#### &#10071; The final project report is due by the end of the day on Tuesday! &#10071;

### Final presentations [[Scoring rubric](./scoring_rubric.pdf)]
* 04/29/21 9-10:50am  
		- Audrey McCombs  
		- Cameron Vannoy + Kaitlyn Barnes + Rachel Rompala
		- Zerui Zhang  
		- Mudith Ekanayake  
		- Mihir Kharate  
		- Kavi Raj Acharya  

### Week 15 (May 3)  

* 05/03/21 8-10am  
		- Jaspreet Rishi  
		- Ceren Ordas  
		- Yawei Li  
		- Bruno do Rosario Petrucci  
		- Ann Murithi  




    * Reading (optional): Soul and Wright 2020. Phylogenetic Comparative Methods: A User’s Guide for Paleontologists. [EcoEvoRxiv Preprints](10.32942/osf.io/ytm5x)


### Week 14 (Apr 27, 29) -- COVID-19 edition
* Topic 17: Phylogenomics and the tree of life [[notes](https://isu-molphyl.github.io/EEOB563-Spring2018/lecture_notes/04_10_18.pdf)].
    * In lieu of the lecture, read: Delsuc et al. 2005.  Phylogenomics and the reconstruction of the tree of life.  [Nature Reviews Genetics.  6: 361-375](https://www.nature.com/articles/nrg1603). (This is old now, but rather classical).
    * Discussion: Simion et al. (2017). A Large and Consistent Phylogenomic Dataset Supports Sponges as the Sister Group to All Other Animals. [Current biology 2: 958-967](https://www.sciencedirect.com/science/article/pii/S0960982217301999).
* Phylogenomics and the tree of life  [[notes](https://isu-molphyl.github.io/EEOB563-Spring2018/lecture_notes/04_10_18.pdf)].
    * In lieu of the lecture, read: Simion et al. (2021). To What Extent Current Limits of Phylogenomics Can Be Overcome?.
    In Scornavacca et al., editors, [Phylogenetics in the Genomic Era, chapter No. 2.1, pp. 2.1:1–2.1:34](https://hal.archives-ouvertes.fr/hal-02535366/document).
    * Discussion (again): Simion et al. (2017). A Large and Consistent Phylogenomic Dataset Supports Sponges as the Sister Group to All Other Animals. [Current biology 2: 958-967](https://www.sciencedirect.com/science/article/pii/S0960982217301999).

		### Week 14 (April 27, 29)
		* Student presentations


* **Revised final project due on 4/28 at 11:59pm (commit to GitHub and email me the link)**

### Final Presentations [[Scoring rubric](./scoring_rubric.pdf)]:
* 04/30: 9:00-10:50am
  * Elizabeth
  * Lauren  
  * Jing
  * Jermaine
  * Zakayo
* 05/02: 9:00-10:50am
  * Shofi
  * Jade
  * Amruta
  * Jordan
  * Dennis
* 05/07: 8:00-9:30am
  * Katelyn
  * Devin
  * Christian
  * Zachary

### Week 11 (Mar 24, 26) -- Coronavirus edition
* Lecture 12: Model selection and model averaging in Likelihood and Bayesian methods.
    * Reading and Discussion: Posada and Buckley 2004.  Model selection and model averaging in phylogenetics:
    advantages of Akaike Information Criterion and Bayesian Approaches over Likelihood Ratio tests. [Systematic Biology, 53:793?808](https://academic.oup.com/sysbio/article/53/5/793/2842928).


* Midterm exam review and [final project](./final_project.md) discussion
    * Be ready to present your final project outline.  Include hypotheses, data, and proposed methods for the project. In addition, create a GitHub/GitLab repository for the final project and send me the
link.

* [Computer Lab 5](https://isu-molphyl.github.io/EEOB563/computer_labs/lab5): Bayesian analysis with MrBayes.
  * [Assignment 6](https://isu-molphyl.github.io/EEOB563-Spring2021/assignments/assignment6.pdf) (**due 04/04**) [[dataset]](https://isu-molphyl.github.io/EEOB563-Spring2021/assignments/hiv.nxs)


Lecture 13 (ha!): Gene trees and species trees.
    * Reading: Allman and Rhodes (2016).  Chapter 13: Gene trees and species trees
    * Discussion: Copetti et al. 2017. Extensive gene tree discordance and hemiplasy shaped the genomes of North American columnar cacti. [Proc Natl Acad Sci U S A: 114: 12003-12008](http://www.pnas.org/   
content/114/45/12003.full).

### Week 12
* Lecture 14: Neutral and adaptive protein evolution [[notes](https://isu-molphyl.github.io/EEOB563-Spring2021/lecture_notes/08_Molecular_evolution.pdf)].
    * Reading: Bielawski and Yang (2000). Statistical methods for detecting molecular adaptation.  [TREE 15(12): 496-503](http://www.cell.com/trends/ecology-evolution/abstract/S0169-5347(00)01994-7).
    * Discussion (leaders _Jade and Zachary_):  Tenaillon et al. (2016). Tempo and mode of genome evolution in a 50,000-generation experiment. [Nature 536: 165-170](https://www.nature.com/articles/         
nature18959).

* [Computer Lab 6](https://isu-molphyl.github.io/EEOB563/computer_labs/lab6): Hypotheses testing with PAML.

### Week 13
* Lecture 15: Timing the evolutionary events [[notes](https://isu-molphyl.github.io/EEOB563-Spring2018/lecture_notes/04_03_18.pdf)].
    * Reading: Sauquet (2013). A practical guide to molecular dating. [C. R. Palevol 12. 355?367](https://www.sciencedirect.com/science/article/pii/S1631068313001097)
    * Discussion: Worobey et al. (2016). 1970s and ?Patient 0? {HIV}-1 genomes illuminate early {HIV}/{AIDS} history in North America. [Nature 539: 98-101](https://www.nature.com/articles/nature19827).

* [Computer Lab 7](https://isu-molphyl.github.io/EEOB563/computer_labs/lab7): Taming the BEAST.

### Week 14

* [Computer Lab 8](https://isu-molphyl.github.io/EEOB563/computer_labs/lab8): BayesTraits.
    * Reading (again): Pagel and Meade (2006).  Bayesian Analysis of Correlated Evolution of Discrete Characters by  Reversible-Jump Markov Chain Monte Carlo.  Am. Nat. 167:808-825.
    * **Final project draft due:** share your GitHub/GitLab address with two assigned reviewers.  Perform the kind of positive, constructive review you would like to get on your own draft.  Prepare your    
reviews by 4/23.  

### Week 15 (Current and Last!)
* Lecture 17: Phylogenomics and the tree of life [[notes](https://isu-molphyl.github.io/EEOB563-Spring2018/lecture_notes/04_10_18.pdf)].
    * Reading: Delsuc et al. 2005.  Phylogenomics and the reconstruction of the tree of life.  [Nature Reviews Genetics.  6: 361-375](https://www.nature.com/articles/nrg1603).
    * Discussion: Simion et al. (2017). A Large and Consistent Phylogenomic Dataset Supports Sponges as the Sister Group to All Other Animals. [Current biology 2: 958-967](https://www.sciencedirect.com/    
science/article/pii/S0960982217301999).

* Lecture 18: Signal vs. noise in phylogenetic reconstruction [[notes](https://isu-molphyl.github.io/EEOB563-Spring2018/lecture_notes/04_10_18.pdf)].
    * Discussion: Philippe et al. 2017. Pitfalls in supermatrix phylogenomics. [European Journal of Taxonomy 283: 1-25](http://www.europeanjournaloftaxonomy.eu/index.php/ejt/article/view/407).


* **Revised final project due on 4/28 at 11:59pm (commit to GitHub and email me the link)**

--->
