---
layout: page
title: EEOB 563 - Schedule
sidebar: Schedule
navigation: 2
icon: <i class='fa fa-calendar'></i>
---

# Schedule

I will update this page before every class with the links to in-class materials and assignments.
To view the planned schedule of course activities, view the class calendar using the button below.

<a href="https://docs.google.com/spreadsheets/d/1okuP20eZHE3TKdtxlqVbEp3xEcYB6wGZh_O1muiiXb0/edit?usp=sharing"><button type="button" class="btn btn-primary">Class Calendar</button></a>

### Week 1 (Jan 17, 19)
* Lecture 1: Introduction to Molecular Phylogenetics [[slides](https://isu-molphyl.github.io/EEOB563-Spring2023/lecture_notes/00_Introduction.pdf)]
	* [Assignment 1](./assignments.md) (**due 1/19**)
* Lecture 2: Trees and characters [[notes](https://isu-molphyl.github.io/EEOB563-Spring2023/lecture_notes/01-phylogenetic_trees.pdf)]
    * [Assignment 1](./assignments.md) is due.
    * Reading: Allman and Rhodes (2016).  Chapter 2: Combinatorics of Trees I.  
    * Discussion: Baum et al. (2005). The Tree-Thinking Challenge.  [Science 310: 979-980](http://science.sciencemag.org/content/310/5750/979.full.pdf).  

### Week 2 (Jan 24, 26)
* Lecture 3: Homology and Sequence Alignment [[notes](https://isu-molphyl.github.io/EEOB563-Spring2023/lecture_notes/02-homology_and_msa.pdf)]
    * Reading: Lemey, Salemi and Vandamme, [Chapter 3: 3.1-3.10](./chapter3.pdf).  
    * Discussion: Fitch (2000). Homology a personal view on some of the problems. [TIG 16: 227](https://doi.org/10.1016/S0168-9525(00)02005-9).
* [Computer Lab 1](https://isu-molphyl.github.io/EEOB563/computer_labs/lab1): Introduction to Unix/Multiple Sequence Alignment (MSA).
    * Complete all [prerequisites](https://isu-molphyl.github.io/EEOB563/computer_labs/lab1) before coming to the class!
    * Post any question you may have on Slack.
    * [Assignment 2](https://isu-molphyl.github.io/EEOB563-Spring2023/assignments/assignment2.pdf) (**due 02/02**)

### Week 3 (Jan 31, Feb 2)
* Lecture 4: Parsimony and cladistics. Optimality criteria and character optimization.  
    * Reading:  Allman and Rhodes (2016).  Chapter 3: Parsimony. [[notes](https://isu-molphyl.github.io/EEOB563-Spring2023/lecture_notes/03_Maximum_parsimony.pdf)]
* Lecture 5: Searching tree space. Measures of character fit. Assessing clade support.  
    * Reading: Allman and Rhodes (2016).  Chapter 9: Tree Space. [[notes](https://isu-molphyl.github.io/EEOB563-Spring2023/lecture_notes/03_Maximum_parsimony.pdf)]
    * Discussion: Baron et al. (2017). A new hypothesis of dinosaur relationships and early dinosaur evolution. [Nature 543: 501-506](https://www.nature.com/articles/nature21700).
    See also [News & Views](https://www.nature.com/articles/543494a), Langer et al. [communication](https://www.nature.com/articles/nature24011), and Baron et al. [reply](https://www.nature.com/articles/nature24012).

### Week 4 (Feb 7, 9)
* [Computer Lab 2](https://isu-molphyl.github.io/EEOB563/computer_labs/lab2): Multiple Sequence Alignment and Parsimony analysis.
    * You may want to install Mafft, PAUP\*, and TNT on your computer [[links](https://isu-molphyl.github.io/EEOB563/links)]. However, we will use HPC-class cluster for the excercises.
    * [Assignment 3](https://isu-molphyl.github.io/EEOB563-Spring2023/assignments/assignment3.pdf) (**due 02/16**)
* Lecture 6: Distance matrix methods. Clustering algorithms. [[notes](https://isu-molphyl.github.io/EEOB563-Spring2023/lecture_notes/04_Distance_methods.pdf)]
    * Reading: Allman and Rhodes (2016).  Chapter 5: Distance Methods.

### Week 5 (Feb 14, 16)  
* Lecture 7: NJ algorithm; Model-based distances. [[notes](https://isu-molphyl.github.io/EEOB563-Spring2023/lecture_notes/04_Distance_methods.pdf)]
    * Reading: (Optional!) Allman and Rhodes (2016).  Chapter 6: Probabilistic Models of DNA Mutations.
    * Discussion: Naxerova et al. (2017). Origins of lymphatic and distant metastases in human colorectal cancer. [Science 357, 55-60](http://science.sciencemag.org/content/357/6346/55).
* [Computer Lab 3](https://isu-molphyl.github.io/EEOB563/computer_labs/lab3): Distance analysis with PAUP, PHYLIP, and FastME.  
    * [Assignment 4](https://isu-molphyl.github.io/EEOB563-Spring2023/assignments/assignment4.pdf) (**due 02/28**)

### Week 6 (Feb 21, 23)
* Lecture 8: Introduction to Maximum Likelihood. [[notes](https://isu-molphyl.github.io/EEOB563-Spring2023/lecture_notes/05_Maximum_likelihood.pdf)]
    * Reading: Allman and Rhodes (2016).  Chapter 8: Maximum Likelihood.
* Lecture 9: Constructing Phylogenetic Trees Using Maximum Likelihood. [[notes](https://isu-molphyl.github.io/EEOB563-Spring2023/lecture_notes/05_Maximum_likelihood.pdf)]
    * Discussion: Basem Al-Shayeb et al. (2020). Clades of huge phages from across Earth’s ecosystems. [Nature 578, 425–431](https://doi.org/10.1038/s41586-020-2007-4).
    If you want to brush up your knowledge on bacteriophages, here is a nice [review](https://doi.org/10.1038/s41586-019-1894-8) from the same issue of Nature.

### Week 7 (Feb 28, Mar 2)
* [Computer Lab 4](https://isu-molphyl.github.io/EEOB563/computer_labs/lab4): Likelihood analysis in RAxML and IQ-Tree.
    * [Assignment 5](https://isu-molphyl.github.io/EEOB563-Spring2023/assignments/assignment5.pdf) (**due 03/09**)
* Lecture 10: Bayes’ theorem and Bayesian methods in phylogenetics. [[notes](https://isu-molphyl.github.io/EEOB563-Spring2023/lecture_notes/06_Bayesian_phylogenetics.pdf)]
    * Reading: Allman and Rhodes (2016).  Chapter 12: Bayesian Inference.

### Week 8 (Mar 7, 9)
* Lecture 11: Applications of Bayesian methods. [[notes](https://isu-molphyl.github.io/EEOB563-Spring2023/lecture_notes/06_Bayesian_phylogenetics.pdf)]
    * Discussion: Williams et al., 2020. Microbial predators form a new supergroup of eukaryotes [Nature, 612:714-719](https://doi.org/10.1038/s41586-022-05511-5)
* [Computer Lab 5](https://isu-molphyl.github.io/EEOB563/computer_labs/lab5): Bayesian analysis with MrBayes.
     * [Assignment 6](https://isu-molphyl.github.io/EEOB563-Spring2023/assignments/assignment6.pdf) (**due 03/23**)[[dataset]](https://isu-molphyl.github.io/EEOB563-Spring2023/assignments/hiv.nxs)

### Spring Break! (Mar 14, 16)
* Don’t forget about your [final project](./final_project.md) outline! (**due 3/23**)

### Week 9 (Mar 21, 23)
* Lecture 12: Model selection and model averaging in Likelihood and Bayesian methods [[notes](https://isu-molphyl.github.io/EEOB563-Spring2023/lecture_notes/07_Model_use.pdf)].
    * Reading: Posada & Buckley (2004).  Model Selection and Model Averaging in Phylogenetics: Advantages of Akaike Information Criterion and Bayesian Approaches Over Likelihood Ratio Tests.
    [Systematic Biology 53: 793–808](https://doi.org/10.1080/10635150490522304).  
	* Alternative (simpler) reading: Kelchner & Thomas (2007). Model use in phylogenetics: nine key questions. [TREE 22: 87-94](https://doi.org/10.1016/j.tree.2006.10.004)
    * Discussion: Abadi et al. 2019. Model selection may not be a mandatory step for
    phylogeny reconstruction. [Nature Communications 10:934](https://doi.org/10.1038/s41467-019-08822-w)  
	* There is a critique in [BioRxiv](https://doi.org/10.1101/849018) if you are interested.  
    * Remember to post at least one comment question in the reading-discussion channel on Slack!  
* Lecture 13: * Midterm exam review and [final project](./final_project.md) discussion
    * Be ready to present your final project outline. Include hypotheses, data, and proposed methods for the project. In addition, create a GitHub repository for the final project and send me the link.

#### &#10071; Midterm exam &#10071;
- Exam will open on Mar 23, after the class. Will cover weeks 1-8
- Exam will close on Mar 28 before the class.
- You will have one 3hr window to complete the exam.


### Week 10 (Mar 28, 30)
* Lecture 14: Neutral and adaptive protein evolution [[notes](https://isu-molphyl.github.io/EEOB563-Spring2023/lecture_notes/08_Molecular_evolution.pdf)].
    * Reading: Vitti et al. (2013). Detecting Natural Selection in Genomic Data. [Annual Review of Genetics 47:97-120](https://doi.org/10.1146/annurev-genet-111212-133526).
    <!-- consider Polygenic adaptation: a unifying framework to understand positive selection -->
    * Discussion:  Tenaillon et al. (2016). Tempo and mode of genome evolution in a 50,000-generation experiment [Nature 536: 165–170](https://doi.org/10.1038/nature18959).
* [Computer Lab 6](https://isu-molphyl.github.io/EEOB563/computer_labs/lab6): Hypotheses testing with PAML.

### Week 11 (Apr 4, 6)
* Lecture 15: Gene trees and species trees [[notes](https://isu-molphyl.github.io/EEOB563-Spring2023/lecture_notes/09_Gene_species_trees.pdf)]. 
    * Overview: Duchêne (2021). Phylogenomics. [Current Biology 31: PR1177-R1181](https://doi.org/10.1016/j.cub.2021.07.039)
    * Reading: Mirarab et al. (2021) Multispecies Coalescent: Theory and Applications in Phylogenetics. [Annu. Rev. Ecol. Evol. Syst. 2021. 52:247–68](https://doi.org/10.1146/annurev-ecolsys-012121-095340)
 <!--    * Reading: Szöllősi et al. (2015). The Inference of Gene Trees with Species Trees. [Systematic Biology 64:e42–e62](https://doi.org/10.1093/sysbio/syu048).  
 -->
    * Discussion: Romiguier et al. (2022). Ant phylogenomics reveals a natural selection hotspot preceding the origin of complex eusociality. [Current Biology 32: 2942-2947.e4](https://doi.org/10.1016/j.cub.2022.05.001)  
* Lecture 16: Phylogenomics and the tree of life  [[notes](https://isu-molphyl.github.io/EEOB563-Spring2023/lecture_notes/10_Phylogenomics.pdf)].  
* Reading: Kapli et al. (2020) Phylogenetic tree building in the genomic age. [Nature Reviews Genetics 21: 428–444](https://doi.org/10.1038/s41576-020-0233-0)
	* Reading (optional): Carter et al. (2023). Estimating phylogenies from genomes: A beginners review of commonly used genomic data in vertebrate phylogenomics. [Journal of Heredity 114:1–13](https://doi.org/10.1093/jhered/esac061)
    * Discussion: Simion et al. (2021). To What Extent Current Limits of Phylogenomics Can Be Overcome?.
		In Scornavacca et al., editors, [Phylogenetics in the Genomic Era, chapter No. 2.1, pp.  2.1:1–2.1:34](https://hal.archives-ouvertes.fr/hal-02535366/document).    

 <!-- * Lecture 17: Signal vs. noise in phylogenetic reconstruction [[notes](https://isu-molphyl.github.io/EEOB563-Spring2018/lecture_notes/04_10_18.pdf)].
    * Discussion: Philippe et al. 2017. Pitfalls in supermatrix phylogenomics. [European Journal of Taxonomy 283: 1-25](http://www.europeanjournaloftaxonomy.eu/index.php/ejt/article/view/407). 
    * Nice resource: Scornavacca et al. (Eds.) (2020). Phylogenetics in the Genomic Era. [Authors open access book, p.p. 1-568](https://hal.inria.fr/hal-02535070v3).
 -->
	
### Week 12 (Apr 11, 13)
* [Computer Lab 7](https://isu-molphyl.github.io/EEOB563/computer_labs/lab7): Phylogenomics.
* Lecture 17: Ancestral State Reconstruction [[notes](https://isu-molphyl.github.io/EEOB563-Spring2023/lecture_notes/11_Ancestral_reconstruction.pdf)].
    * Reading: Joy et al. 2016. Ancestral reconstruction. [PLoS Comput. Biol. 12(7): e1004763](https://doi.org/10.1371/journal.pcbi.1004763)
    * Discussion: Starr et al. (2017). Alternative evolutionary histories in the sequence space of an ancient protein. [Nature 549:409-413](https://doi.org/10.1038/nature23902)
 <!--   * Crisp and Cook (2005). Do early branching lineages signify ancestral traits? [Trends Ecol. Evol. 20:122-128](https://doi.org/10.1016/j.tree.2004.11.010)
 -->

### Week 13 (Apr 18, 20)
* Lecture 18: Phylogenetic comparative methods [[notes](https://isu-molphyl.github.io/EEOB563-Spring2023/lecture_notes/12_Phylogenetic_comparative_methods)].
    * Reading: O’Meara (2012).  Evolutionary Inferences from Phylogenies: A Review of Methods. [Annu. Rev. Ecol. Evol. Syst. 43:267–285](https://doi.org/10.1146/annurev-ecolsys-110411-160331).  
<!-- 
    * Reading: Uyeda et al. 2018. Rethinking phylogenetic comparative methods. [Systematic Biology 67: 1091–1109](https://doi.org/10.1093/sysbio/syy031).
    * Reading (optional): Soul and Wright 2020. Phylogenetic Comparative Methods: A User’s Guide for Paleontologists. [EcoEvoRxiv Preprints](10.32942/osf.io/ytm5x) 
-->
    * Discussion: Watts et al. 2016. Ritual human sacrifice promoted and sustained the evolution of stratified societies. [Nature 532: 228-231](https://www.nature.com/articles/nature17159).
* [Computer Lab 8](https://isu-molphyl.github.io/EEOB563/computer_labs/lab8): BayesTraits.
    * Introductory reading: Pagel and Meade (2006).  Bayesian Analysis of Correlated Evolution of Discrete Characters by  Reversible-Jump Markov Chain Monte Carlo.  [Am. Nat. 167:808-825](https://doi.org/10.1086/503444).
    * **Final project draft due:** share your GitHub/GitLab address with two assigned reviewers.  Perform the kind of positive, constructive review you would like to get on your own draft.  Prepare your reviews by 4/25.  

### Week 14 (Apr 25, 27)
* Lecture 18: Timing the evolutionary events [[notes](https://isu-molphyl.github.io/EEOB563-Spring2023/lecture_notes/13_Molecular_clocks.pdf)].
    * Reading: Bromham et al. (2018). Bayesian molecular dating: opening up the black box. [Biological Reviews, 93: 1165-1191](https://doi.org/10.1111/brv.12390)
    * Discussion: Worobey et al. (2016). 1970s and "Patient 0" {HIV}-1 genomes illuminate early HIV/AIDS history in North America. [Nature 539: 98-101](https://www.nature.com/articles/nature19827).  
<!-- * Discussion: Worobey (2008). Direct evidence of extensive diversity of HIV-1 in Kinshasa by 1960.
    [Nature 455: 661–664](https://doi.org/10.1038/nature07390). 
      * Remember to post at least one comment/question on either paper in the reading-discussion channel on Slack and answer/comment on another post.
-->
* [Computer Lab 9](https://isu-molphyl.github.io/EEOB563/computer_labs/lab7): Taming the BEAST.
    * Reading (optional): Sauquet (2013). A practical guide to molecular dating. [C. R. Palevol 12. 355?367](https://www.sciencedirect.com/science/article/pii/S1631068313001097)

#### &#10071; The final project report is due by the end of the day on Sunday! Submit the link to your GitHub repository on Canvas &#10071;

<!--
### Week 15 (May 2, 4): Final presentations  
[[Scoring rubric](./scoring_rubric.pdf)]
* 05/02: 9:00-10:45am
  * Elizabeth
  * Lauren  
  * Jing
  * Jermaine
  * Zakayo
* 05/04: 9:00-10:45am
  * Shofi
  * Jade
  * Amruta
  * Jordan
  * Dennis


    * 


--->
